src/main/java/com/bakdata/kafka/KafkaProducerApplication.java

```diff
@@ -85,20 +92,29 @@ public void run() {
     protected Properties createKafkaProperties() {
         final Properties kafkaConfig = new Properties();
 
-        kafkaConfig.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, SpecificAvroSerializer.class);
-        kafkaConfig.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, SpecificAvroSerializer.class);
         // exactly once and order
         kafkaConfig.put(ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION, 1);
         kafkaConfig.setProperty(ProducerConfig.ACKS_CONFIG, "all");
 
         // compression
         kafkaConfig.setProperty(ProducerConfig.COMPRESSION_TYPE_CONFIG, "gzip");
 
-        kafkaConfig.setProperty(AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, this.getSchemaRegistryUrl());
+        this.configureDefaultSerializer(kafkaConfig);
         kafkaConfig.setProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, this.brokers);
         return kafkaConfig;
     }
 
+    private void configureDefaultSerializer(final Properties kafkaConfig) {
+        if (nonNull(this.schemaRegistryUrl)) {
```

Comment:
why not just `this.schemaRegistryUrl == null`? (caution, I inverted the condition)